### PR TITLE
path.Clean paths in GlobalEnvParams and remove unnecessary path.Join

### PR DESCRIPTION
**Pull request by bruceauyeung**
**What this PR does / why we need it**:

1. clean all paths in `GlobalEnvParams`
1. remove unnecessary path.Join call in `pki.go`
2. fix some typos and comment errors

Signed-off-by: bruceauyeung <ouyang.qinhua@zte.com.cn>

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

File: cmd/kubeadm/app/master/pki.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ func CreatePKIAssets(cfg *kubeadmapi.MasterConfiguration) (*rsa.PrivateKey, *x50
 	}
 	altNames.DNSNames = append(altNames.DNSNames, cfg.API.ExternalDNSNames...)
 
-	pkiPath := path.Join(kubeadmapi.GlobalEnvParams.HostPKIPath)
+	pkiPath := kubeadmapi.GlobalEnvParams.HostPKIPath
```
Comment (jbeda) [inline cmd/kubeadm/app/master/pki.go]: This isn't equivalent.  `path.Join` will also clean the path.  If you want to remove the Join, you should replace it with `path.Clean`.  See: https://play.golang.org/p/jLUuVGBBSy

**Review by jbeda — CHANGES_REQUESTED** [review]

**Comment by jbeda** [comment]
@k8s-bot cvm gke e2e test this
@k8s-bot cvm gke e2e test this
Flake #35913

**Pull request merged**